apps/deploy-web/src/queries/useAnonymousUserQuery.ts

```diff
@@ -0,0 +1,43 @@
+import { useState } from "react";
+import axios from "axios";
+
+import { useWhen } from "@src/hooks/useWhen";
+import { BASE_API_URL } from "@src/utils/constants";
+
+export interface ApiUserOutput {
+  id: string;
+  userId?: string;
+  username?: string;
+  email?: string;
+  emailVerified: boolean;
+  stripeCustomerId?: string;
+  bio?: string;
+  subscribedToNewsletter: boolean;
+  youtubeUsername?: string;
+  twitterUsername?: string;
+  githubUsername?: string;
+}
+
+export const anonymousUsersHttp = axios.create({
+  baseURL: `${BASE_API_URL}/v1/anonymous-users`
+});
+
+const createAnonymousUser = async () => (await anonymousUsersHttp.post<ApiUserOutput>("")).data;
+const getAnonymousUser = async (id: string) => (await anonymousUsersHttp.get<ApiUserOutput>(id)).data;
+
+let userAsPromised: Promise<ApiUserOutput>;
+const findOrCreateAnonymousUser = async (id?: string) => {
+  userAsPromised = userAsPromised || (id ? getAnonymousUser(id) : createAnonymousUser());
+  return await userAsPromised;
+};
+
+export function useAnonymousUserQuery(id?: string, options?: { enabled?: boolean }) {
```

Comment:
So since this is not called anywhere, it won't affect the state of the console?

Reply:
not sure I follow

Reply:
When we merge into main and deploy, it won't be called anywhere right?

Reply:
it would actually. It's not hidden behind any flags

Reply:
Is this what we want? Since the feature is not production ready

Reply:
nope, however this is WIP. Even the current chunk of work is not really mergeable. I need to test more, make sure it doesn't affect anything. I do not plan to add new features before merge but I want to test more and adjust if needed. Will disable newly added functionality for sure. thanks for the heads up!